### ISSUE TITLE
Tangent never undefined

### ISSUE THREAD

**Issue by FredericMrozinski** [davidedc/Algebrite#97]
When entering 'float(tan(pi/2))' the result is a big numeric result. Similarly with other spots at which the tangent should be undefined. Maybe output an error message instead?

**Comment by davidedc** [comment]
interesting. Yes the standard c++ and JS functions return the same result see http://cpp.sh/6rlmq and ```Math.tan((Math.PI)/2)``` in JS. We could work around that by comparing with Math definition of PI and give an error or return something.

If we return something like... undefined... that _might_ be complicated to harmonise... TBC

**Comment by FredericMrozinski** [comment]
Oh yes, indeed, it even happens with C++. Interesting, I didn't know that! So maybe for the symbolic simplification, we could add another "when"-case to the tangent file. Something like

`when 90, 270:
      stop("tangent not defined at that point")`

For the part that calculates the tangent numerically, between line 25 and 26 something like:

`if(Math.abs((p1.d / Math.PI) % 1)==0.5)
push_double(Number.NaN)`

I found this solution to work for almost all points at which the tangent is undefined. Only for very few points this doesn't do it (because those few evaluate to e.g. 0.499999....). Maybe then a tolerance thing like:

`if(Math.abs(Math.abs((p1.d / Math.PI) % 1)-0.5) <= 1e-6)
push_double(Number.NaN)`

**Comment by davidedc** [comment]
yes "stop" could be a way... one can't recover from that though, which could be annoying. The other way is to propagate special values such as (a variant of?) NaN but then that could be complicated in other ways. Let's ponder this one a little more.

**Comment by davidedc** [comment]
Wolfram Alpha gives "complex infinity" as result.
![image](https://user-images.githubusercontent.com/769568/66816748-7f9e6100-ef32-11e9-90f2-a61614cf0300.png)
